packages/monaco/src/browser/monaco-editor.ts

```diff
@@ -406,6 +404,14 @@ export class MonacoEditor extends MonacoEditorServices implements TextEditor {
         return this.onMouseDownEmitter.event;
     }
 
+    get onDidChangeReadOnly(): Event<boolean | MarkdownString> {
+        return this.document.onDidChangeReadOnly;
+    }
+
+    get onEncodingChanged(): Event<string> {
+        return this.document.onDidChangeEncoding;
+    }
```

Comment:
There are a number of changes like this, of constructor-initialized field access -> getter. I imagine that falls under the heading of changes 'not to access fields before initialization.' I also tried reverting some of these and ran into no build or runtime problems. Is this also something that would be a problem if we switched to define-property semantics?

The reason I mention this is that if there's no build failure (and no lint rule) that detects these, new code very well may just bring in new instances, so if it's a problem, or something we want to make sure avoids becoming a problem, we may need some alternate means.